### PR TITLE
Add pending `qa-docs` `0.1` dev and new changes

### LINKED ISSUE #2601: Add framework changes to `qa-docs` `0.1`
Some changes have been made in the framework. We need to update the `qa-docs` terms and behavior:

- [x] #2588
- [x] #2590 
- [x] #2591 
- [x] #2600
- [x] #2589

### PR DESCRIPTION
**Pull request by roronoasins**
|Related issues|
|---|
|Close #2601 #1864 #2075|

This PR adds the missing development of `qa-docs` `0.1`, which mostly was merged previously so `qa-ctl` could read the modules' documentation. The pending development issues that could not be closed because of #2196 will be closed with this PR, and #2196 will be sent to the icebox.

Also, the tool has been adapted to the new framework changes where different modules in different suites can have the 
same name, so `qa-ctl` has been updated too. Besides, the module's terminology has been reworked.

The new changes for both `qa-docs` and `qa-ctl` are as follows:

### Added

- Added output format selection.

### Changed

- Change the docker deployment. 
- Adapt the tool to some framework changes. ([#2605](https://github.com/wazuh/wazuh-qa/pull/2605), [#2640](https://github.com/wazuh/wazuh-qa/pull/2640))
- Change the schema since framework changes. ([#2590](https://github.com/wazuh/wazuh-qa/issues/2590))

### Added

- Added new parameters related to the run method. ([#2640](https://github.com/wazuh/wazuh-qa/pull/2640))

### Changed

- Changed how the tool uses `qa-docs`. ([#2640](https://github.com/wazuh/wazuh-qa/pull/2640))

# Checks

## Jenkins checking

<table>
<thead>
<tr>
<th style="width: 175px;">Test</th>
<th style="width: 499px;">Result</th>
</tr>
</thead>
<tbody>
<tr>
<td style="width: 175px;"><code>Analysisd IT - Tier 0 - manager</code></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/23000">🟢</a></td>
</tr>
</tr>
<tr>
<td style="width: 175px;"><code>Analysisd IT - Tier 1 - manager</code></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/23001/">🟢</a></td>
</tr>
</tr>
<tr>
<td style="width: 175px;"><code>AR IT - Tier 0 - manager</code></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/23002">🟢</a></td>
</tr>
</tr>
<tr>
<td style="width: 175px;"><code>AR IT - Tier 1 - centos_agent</code></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/23003">🟢</a></td>
</tr>
</tr>
<tr>
<td style="width: 175px;"><code>Authd IT - Tier 0 - manager</code></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/23004">🟢</a></td>
</tr>
</tr>
<tr>
<td style="width: 175px;"><code>Authd IT - Tier 1 - manager</code></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/23005">🟢</a></td>
</tr>
</tr>
<tr>
<td style="width: 175px;"><code>FIM IT - Tier 0 - centos_agent</code></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/23006">🟢</a></td>
</tr>
</tr>
<tr>
<td style="width: 175px;"><code>FIM IT - Tier 0 - manager</code></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/23007">🟢</a></td>
</tr>
</tr>
<tr>
<td style="width: 175px;"><code>FIM IT - Tier 1 - centos_agent</code></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/23008">🟢</a></td>
</tr>
</tr>
<tr>
<td style="width: 175px;"><code>FIM IT - Tier 1 - manager</code></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/23009">🟢</a></td>
</tr>
</tr>
<tr>
<td style="width: 175px;"><code>Logtest IT - Tier 0 - manager</code></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/23010">🟢</a></td>
</tr>
</tr>
<tr>
<td style="width: 175px;"><code>Remoted IT - Tier 0 - manager</code></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/23011">🟢</a></td>
</tr>
</tr>
<tr>
<td style="width: 175px;"><code>Remoted IT - Tier 1 - manager</code></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/23012">🟢</a></td>
</tr>
</tr>
<tr>
<td style="width: 175px;"><code>Rids IT - Tier 0 - manager</code></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/23013">🟢</a></td>
</tr>
</tr>
<tr>
<td style="width: 175px;"><code>Rootcheck IT - Tier 0 - manager</code></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/23014/">🟢</a></td>
</tr>
</tr>
<tr>
<td style="width: 175px;"><code>Rootcheck IT - Tier 1 - manager</code></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/23015">🟢</a></td>
</tr>
</tr>
<tr>
<td style="width: 175px;"><code>Vul. Det. IT - Tier 0 - manager</code></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/23029">🟢</a></td>
</tr>
</tr>
<tr>
<td style="width: 175px;"><code>Vul. Det. IT - Tier 1 - manager</code></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/23030">🟢</a></td>
</tr>
</tr>
<tr>
<td style="width: 175px;"><code>Wazuh DB IT - Tier 0 - manager</code></td>
<td style="width: 103px;"><a href="https://ci.wazuh.info/job/Test_integration/23018">🟢</a></td>
</tr>
</tr>
</tbody>
</table>

### CODE REVIEW

File: tests/integration/test_vulnerability_detector/test_general_settings/test_enabled.py [hunk 1]
```diff
@@ -11,10 +11,16 @@
        is working correctly. This option is located in its corresponding section of
        the `ossec.conf` file and allows enabling or disabling this module.
 
+<<<<<<< HEAD
```
Comment (jmv74211) [inline tests/integration/test_vulnerability_detector/test_general_settings/test_enabled.py:14]: Conflicts have not been properly resolved

File: tests/integration/test_vulnerability_detector/test_general_settings/test_interval.py [hunk 2]
```diff
@@ -11,10 +11,16 @@
        is working correctly. This option is located in its corresponding section of
        the `ossec.conf` file and allows to define the time between scans.
 
+<<<<<<< HEAD
```
Comment (jmv74211) [inline tests/integration/test_vulnerability_detector/test_general_settings/test_interval.py:14]: Conflicts have not been properly resolved

File: tests/integration/test_vulnerability_detector/test_general_settings/test_min_full_scan_interval.py [hunk 3]
```diff
@@ -12,10 +12,16 @@
        and allows to define the minimum time before performing a full scan even if the feed was updated. The extended
        behavior also checks that the full scan is performed as expected.
 
+<<<<<<< HEAD
```
Comment (jmv74211) [inline tests/integration/test_vulnerability_detector/test_general_settings/test_min_full_scan_interval.py:15]: Conflicts have not been properly resolved

File: tests/integration/test_vulnerability_detector/test_general_settings/test_run_on_start.py [hunk 4]
```diff
@@ -1,20 +1,30 @@
...
-copyright: Copyright (C) 2015-2021, Wazuh Inc.
+copyright: Copyright (C) 2015-2022, Wazuh Inc.
 
            Created by Wazuh, Inc. <info@wazuh.com>.
 
            This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
 
 type: integration
 
+<<<<<<< HEAD
```
Comment (jmv74211) [inline tests/integration/test_vulnerability_detector/test_general_settings/test_run_on_start.py:10]: Conflicts have not been properly resolved

**Review by jmv74211 — CHANGES_REQUESTED** [review]

File: tests/integration/test_vulnerability_detector/test_general_settings/test_run_on_start.py (hunk 4 above)
Comment (roronoasins) [reply tests/integration/test_vulnerability_detector/test_general_settings/test_run_on_start.py]: Solved https://github.com/wazuh/wazuh-qa/pull/2649/commits/71e4587f39df3f93b1e16f8387b2c14fb5bc217b

File: tests/integration/test_vulnerability_detector/test_general_settings/test_min_full_scan_interval.py (hunk 3 above)
Comment (roronoasins) [reply tests/integration/test_vulnerability_detector/test_general_settings/test_min_full_scan_interval.py]: Solved https://github.com/wazuh/wazuh-qa/pull/2649/commits/71e4587f39df3f93b1e16f8387b2c14fb5bc217b

File: tests/integration/test_vulnerability_detector/test_general_settings/test_interval.py (hunk 2 above)
Comment (roronoasins) [reply tests/integration/test_vulnerability_detector/test_general_settings/test_interval.py]: Solved https://github.com/wazuh/wazuh-qa/pull/2649/commits/71e4587f39df3f93b1e16f8387b2c14fb5bc217b

File: tests/integration/test_vulnerability_detector/test_general_settings/test_enabled.py (hunk 1 above)
Comment (roronoasins) [reply tests/integration/test_vulnerability_detector/test_general_settings/test_enabled.py]: Solved https://github.com/wazuh/wazuh-qa/pull/2649/commits/71e4587f39df3f93b1e16f8387b2c14fb5bc217b

File: tests/integration/test_remoted/test_configuration/test_basic_configuration_rids_closing_time_valid.py [hunk 5]
```diff
@@ -108,6 +100,11 @@ def test_rids_closing_time_valid(get_configuration, configure_environment, resta
 
     wazuh_min_version: 4.2.0
 
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply tests/integration/test_remoted/test_configuration/test_basic_configuration_rids_closing_time_valid.py]: Solved https://github.com/wazuh/wazuh-qa/pull/2649/commits/71e4587f39df3f93b1e16f8387b2c14fb5bc217b

File: tests/integration/test_remoted/test_configuration/test_basic_configuration_local_ip_valid.py [hunk 6]
```diff
@@ -107,6 +99,11 @@ def test_local_ip_valid(get_configuration, configure_environment, restart_remote
 
     wazuh_min_version: 4.2.0
 
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply tests/integration/test_remoted/test_configuration/test_basic_configuration_local_ip_valid.py]: Solved https://github.com/wazuh/wazuh-qa/pull/2649/commits/71e4587f39df3f93b1e16f8387b2c14fb5bc217b

File: tests/integration/test_logcollector/test_command_monitoring/test_command_execution_dbg.py [hunk 7]
```diff
@@ -224,22 +231,25 @@ def test_command_execution_dbg(get_files_list, create_file_structure_module, con
...
     assertions:
         - Verify that the debug 'running' event is generated when running the command set in the 'command' tag.
         - Verify that the debug 'reading command' event is generated when running the related command.
         - Verify that the debug 'lines' event is generated when running the related command.
+
     input_description: A configuration template (test_command_execution) is contained in an external
-                       YAML file (wazuh_command_conf.yaml), which includes configuration settings for
-                       the 'wazuh-logcollector' daemon and, it is combined with the test cases
-                       (log formats and commands to run) defined in the module.
+                        YAML file (wazuh_command_conf.yaml), which includes configuration settings for
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply tests/integration/test_logcollector/test_command_monitoring/test_command_execution_dbg.py]: Solved https://github.com/wazuh/wazuh-qa/pull/2649/commits/71e4587f39df3f93b1e16f8387b2c14fb5bc217b

File: tests/integration/test_logcollector/test_command_monitoring/test_command_execution_dbg.py [hunk 8]
```diff
@@ -193,18 +196,22 @@ def get_files_list():
...
-                 special characteristics. For this purpose, the test will configure the logcollector to run
-                 a command, setting it in the 'command' tag and using the 'command' and 'full_command' log
-                 formats. The properties of that command can be, for example, a non-existent command or one
-                 that includes special characters. Once the logcollector has started, it will wait for the
-                 'running' event that indicates that the command has been executed. Finally, the test
-                 will verify that the debug 'read N lines' event is generated, this event indicates the number
-                 of lines read from the command run. Depending on test case, the test also will verify that
-                 the debug event 'reading command' is generated, this event includes the output of the command
-                 run, and its alias if it is set in the 'alias' tag.
+                    special characteristics. For this purpose, the test will configure the logcollector to run
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply tests/integration/test_logcollector/test_command_monitoring/test_command_execution_dbg.py]: Solved https://github.com/wazuh/wazuh-qa/pull/2649/commits/71e4587f39df3f93b1e16f8387b2c14fb5bc217b

File: tests/integration/test_enrollment/test_agentd_enrollment.py [hunk 9]
```diff
@@ -113,8 +120,16 @@ def test_agentd_enrollment(configure_environment, override_wazuh_conf, get_curre
             "Check that different configuration generates the adequate enrollment message or the corresponding error
             log. The configuration, keys, and password files will be written with the different scenarios described
             in the test cases. After this, Agentd is started to wait for the expected result."
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply tests/integration/test_enrollment/test_agentd_enrollment.py]: Solved https://github.com/wazuh/wazuh-qa/pull/2649/commits/71e4587f39df3f93b1e16f8387b2c14fb5bc217b

File: tests/integration/test_enrollment/test_agent_auth_enrollment.py [hunk 10]
```diff
@@ -1,46 +1,53 @@
...
+copyright: Copyright (C) 2015-2021, Wazuh Inc.
+
+           Created by Wazuh, Inc. <info@wazuh.com>.
+
+           This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
 type: integration
+
 brief: This module verifies the correct behavior of the agent-auth enrollment tool under different configurations
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply tests/integration/test_enrollment/test_agent_auth_enrollment.py]: Solved https://github.com/wazuh/wazuh-qa/pull/2649/commits/71e4587f39df3f93b1e16f8387b2c14fb5bc217b

File: deps/wazuh_testing/wazuh_testing/scripts/qa_ctl.py [hunk 11]
```diff
@@ -282,8 +342,24 @@ def get_script_parameters():
                         help='Config generation mode. The test data will be processed and the configuration will be '
                              'generated without running anything.')
 
-    parser.add_argument('--run', '-r', type=str, action='store', required=False, nargs='+', dest='run_test',
-                        help='Independent run method. Specify a test or a list of tests to be run.')
+    parser.add_argument('--run', '-r', action='store_true',
+                        help='Independent run method. The tests that the use specified will be run.')
```
[the remark this answers is not in the document]
Comment (roronoasins) [reply deps/wazuh_testing/wazuh_testing/scripts/qa_ctl.py]: Solved https://github.com/wazuh/wazuh-qa/pull/2649/commits/12e3f6932f90e08add6148e7dc7b9d87a8bd8230

**Comment by fedepacher** [comment]
<table>
<thead>
<tr>
<th style="width: 175px;">Test</th>
<th style="width: 499px;">Result</th>
</tr>
</thead>
<tbody>
<tr>
<td style="width: 175px;"><code>test_feeds/test_cpe_indexing - manager</code></td>
<td style="width: 103px;"><a>🟢</a></td>
</tr>
<tr>
<td style="width: 175px;"><code>test_feeds/test_download_feeds - manager</code></td>
<td style="width: 103px;"><a>🟢</a></td>
</tr>
<tr>
<td style="width: 175px;"><code>test_feeds/test_duplicate_feeds - manager</code></td>
<td style="width: 103px;"><a></a></td>
</tr>
<tr>
<td style="width: 175px;"><code>test_feeds/test_import_invalid_feed_type - manager</code></td>
<td style="width: 103px;"><a>🟢</a></td>
</tr>
<tr>
<td style="width: 175px;"><code>test_feeds/test_validate_feed_content - manager</code></td>
<td style="width: 103px;"><a></a></td>
</tr>
<tr>
<td style="width: 175px;"><code>test_general_settings/test_enabled - manager</code></td>
<td style="width: 103px;"><a></a></td>
</tr>
<tr>
<td style="width: 175px;"><code>test_general_settings/test_interval - manager</code></td>
<td style="width: 103px;"><a></a></td>
</tr>
<tr>
<td style="width: 175px;"><code>test_general_settings/test_min_full_scan_interval - manager</code></td>
<td style="width: 103px;"><a></a></td>
</tr>
<tr>
<td style="width: 175px;"><code>test_general_settings/test_retry_interval - manager</code></td>
<td style="width: 103px;"><a></a></td>
</tr>
<tr>
<td style="width: 175px;"><code>test_general_settings/test_run_on_start - manager</code></td>
<td style="width: 103px;"><a></a></td>
</tr>
<tr>
<td style="width: 175px;"><code>test_providers/test_enabled - manager</code></td>
<td style="width: 103px;"><a></a></td>
</tr>
<tr>
<td style="width: 175px;"><code>test_providers/test_missing_os - manager</code></td>
<td style="width: 103px;"><a></a></td>
</tr>
<tr>
<td style="width: 175px;"><code>test_providers/test_multiple_provider_feeds - manager</code></td>
<td style="width: 103px;"><a></a></td>
</tr>
<tr>
<td style="width: 175px;"><code>test_providers/test_os - manager</code></td>
<td style="width: 103px;"><a></a></td>
</tr>
<tr>
<td style="width: 175px;"><code>test_providers/test_update_from_year - manager</code></td>
<td style="width: 103px;"><a></a></td>
</tr>
<tr>
<td style="width: 175px;"><code>test_providers/test_update_interval - manager</code></td>
<td style="width: 103px;"><a></a></td>
</tr>
</tbody>
</table>

**Review by jmv74211 — APPROVED** [review]

**Pull request merged**